### PR TITLE
Import WLS data for optical photons

### PR DESCRIPTION
**Pull request by stognini**
This PR is tied to issue #886 and imports optical photon wavelength shifting (WLS) data from Geant4.

### CODE REVIEW

File: app/celer-dump-data.cc [hunk 1]
```diff
@@ -702,6 +702,17 @@ void print_optical_material_data(ImportData::ImportOpticalMap const& iom)
...
     cout << endl;
+    cout << "\n## WLS";
+    cout << header;
+    for (auto const& [mid, val] : iom)
+    {
+        auto const& wls = val.wls;
+        cout << POM_STREAM_SCALAR(mid, wls, mean_num_photons, IU::unitless);
+        cout << POM_STREAM_SCALAR(mid, wls, time_constant, IU::time);
+        cout << POM_STREAM_VECTOR(mid, wls, absorption_length, IU::len);
+        cout << POM_STREAM_VECTOR(mid, wls, component, IU::mev);
```
Comment (whokion) [inline app/celer-dump-data.cc]: I find that the WLS component is unitless (it is a distribution of relative population along the re-emission energy axis and is used to build an inverse CDF to sample the emitted photon energy) - sorry about a wrong guess during our discussion at the hackaton.

File: src/celeritas/io/ImportOpticalMaterial.hh [hunk 2]
```diff
@@ -147,6 +147,28 @@ struct ImportOpticalProperty
...
+    ImportPhysicsVector component;  //!< Emission spectrum [MeV, MeV]
+
+    //! Whether all data are assigned and valid
+    explicit operator bool() const
+    {
+        return mean_num_photons > 0 && time_constant > 0
+               && static_cast<bool>(absorption_length)
+               && static_cast<bool>(component)
+               && absorption_length.vector_type == ImportPhysicsVectorType::free
+               && absorption_length.vector_type == component.vector_type;
```
Comment (whokion) [inline src/celeritas/io/ImportOpticalMaterial.hh]:
```suggestion
               && component.vector_type == absorption_length.vector_type;
```
Not sure which is the better for readability.

**Review by whokion — APPROVED** [review]
Thanks @stognini for adding WLS material properties.  All look good except the unit for the WLS component which is unitless, which need to be changed.

File: app/celer-dump-data.cc (hunk 1 above)
Comment (stognini) [reply app/celer-dump-data.cc]: Ohh I see. Thanks, fixing that now.

File: src/celeritas/io/ImportOpticalMaterial.hh (hunk 2 above)
Comment (stognini) [reply src/celeritas/io/ImportOpticalMaterial.hh]: I like your suggestion.

**Review by sethrj — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**